tutorial.md

```diff
@@ -0,0 +1,1140 @@
+      [ ("M-S-z", spawn "xscreensaver-command -lock")
+      , ("M-S-=", unGrab *> spawn "scrot -s"        )
+      , ("M-]"  , spawn "firefox"                   )
+      ]
+```
+
+But we also wanted to add magnification, right?  Luckily for us, there's
+a module for that as well!  It's called [XMonad.Layout.Magnifier].
+Again, take a look at the documentation yourself before reading on—see
+if you can reason out what to do for yourself.  Let's pick the
+`magnifiercz'` modifier from the library; it magnifies a window by a
+given amount, but only if it is a stack window.  Add it to your three
+column layout thusly:
+
+``` haskell
+myLayout = tiled ||| Mirror tiled ||| Full ||| threeCol
+  where
+    threeCol = magnifiercz' 1.3 $ ThreeColMid nmaster delta ratio
+    tiled    = Tall nmaster delta ratio
+    nmaster  = 1      -- Default number of windows in the master pane
+    ratio    = 1/2    -- Default proportion of screen occupied by master pane
+    delta    = 3/100  -- Percent of screen to increment by when resizing panes
+```
+
+Don't forget to import the module!
+
+You can think of the `$` here as putting everything into parentheses
+from the dollar to the end of the line.  If you don't like that you can
+also write
+
+``` haskell
+  threeCol = magnifiercz' 1.3 (ThreeColMid nmaster delta ratio)
+```
+
+instead.
+
+That's it!  Now we have a perfectly functioning three column layout with
+a magnified stack.  If you compare this with the starting screenshots,
+you will see that that's exactly the behaviour we wanted!
+
+A last thing that that's worth knowing about are so-called
+"combinators"—at least we call them that, we can't tell you what to do.
+These are function that compose with the `xmonad` function and add a lot
+of hooks and other things for you (trying to achieve a specific goal),
+so you don't have to do all the manual work yourself.  For example,
+xmonad—by default—is only [ICCCM] compliant.  Nowadays, however, a lot
+of programs (including many compositors) expect the window manager to
+_also_ be [EWMH] compliant.  So let's save ourselves a lot of future
+trouble and add that to xmonad straight away!
+
+This functionality is to be found in the [XMonad.Hooks.EwmhDesktops]
+module, so let's import it:
+
+``` haskell
+  import XMonad.Hooks.EwmhDesktops
+```
+
+We might also consider using the `ewmhFullscreen` combinator.  By
+default, a "fullscreened" application is still bound by its window
+dimensions; this means that if the window occupies half of the screen
+before it was fullscreened, it will also do so afterwards.  Some people
+(me included) really like this behaviour, as applications thinking
+they're in fullscreen mode tend to remove a lot of clutter (looking at
+you, Firefox).  However, because a lot of people explicitly do not want
+this effect (and some applications, like chromium, will misbehave and
+need some [Hacks] to make this work), we will also add the relevant
+function to get "proper" fullscreen behaviour here.
```

Comment:
The link of `Hacks` is broken. I think you are talking about the X.U.Hacks from the git version.

Reply:
I am talking about X.U.Hacks, but since this tutorial will come out with the new release, I've already switched it to the appropriate Hackage address (which currently doesn't exist) so we can't forget